core/app/models/spree/order.rb

```diff
@@ -516,8 +517,9 @@ def apply_free_shipping_promotions
     # e.g. customer goes back from payment step and changes order items
     def ensure_updated_shipments
       if !completed? && shipments.all?(&:pending?)
-        self.shipments.destroy_all
+        #self.shipments.destroy_all
         self.update_column(:shipment_total, 0)
+        #create_proposed_shipments #should I be doing this here?
```

Comment:
Great. Getting that working is my next step then. I also hope I can get rid of setting `shipment_total` too